File: C#/102. Binary Tree Level Order Traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<IList<int>> LevelOrder(TreeNode root) {
        List<IList<int>> list = new List<IList<int>>();
        solution(0,list,root);
        return list;
    }
    public void solution(int lev, List<IList<int>> list, TreeNode root){
        if(root==null){return;}
        if(lev==list.Count){
            List<int> l1 = new List<int>();
            l1.Add(root.val);
            list.Add(l1);
        }
        else{
            list[lev].Add(root.val);
        }
        solution(lev+1,list,root.left);
        solution(lev+1,list,root.right);
    }
}

File: JavaScript/102. Binary Tree Level Order Traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    var list = new Array();
    solution(0,list,root);
    return list;
};
var solution = function(lev, list, root){
    if(root==null){return;}
    if(lev==list.length){
        var l1 = new Array();
        l1.push(root.val);
        list.push(l1);
    }
    else{
        list[lev].push(root.val);
    }
    solution(lev+1,list,root.left);
    solution(lev+1,list,root.right);
};
